File: frontend/src/types/auctionInfo.ts
import { AuctionBidDto } from "./auctionBid";
import { Car, CarListItemDto } from "./car";
import { UserShortInfo } from "./user";

export interface AuctionInfo {
  id: string;
  carId: string;
  startPrice: number;
  buyoutPrice: number;
  startDate: Date;
  endDate: Date;
  isActive: boolean;
  car: Car | null;
}

export interface AuctionInfoDto {
  id: string;
  carId: string;
  startPrice: number;
  buyoutPrice: number;
  startDate: Date;
  endDate: Date;
  maxBid: AuctionBidDto | null;
  isActive: boolean;
  owner: UserShortInfo;
  car: Car | null;
}
export interface AuctionInfoCreateRequest {
  carId: string;
  startPrice: number;
  buyoutPrice: number;
  startDate: Date;
  endDate: Date;
}

export interface AuctionInfoListItemDto {
  id: string;
  carId: string;
  startPrice: number;
  buyoutPrice: number;
  startDate: Date;
  endDate: Date;
  isActive: boolean;
  car: CarListItemDto | null;
  maxBid: AuctionBidDto | null;
}

File: backend/Core/Domain/Entities/AuctionInfo.cs
namespace Core.Domain.Entities
{
    public class AuctionInfo
    {
        public Guid Id { get; set; }
        public Guid CarId { get; set; }
        public decimal StartPrice { get; set; }
        public decimal BuyoutPrice { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public bool IsActive { get; set; } = true;
        public Car? Car { get; set; }
        public ICollection<AuctionBid>? AuctionBids { get; set; }
    }
}